src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/privacyGroup/PrivacyGroupAcceptanceTest.java

```diff
@@ -0,0 +1,165 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.acceptance.send.receive.privacyGroup;
+
+import static io.vertx.core.Vertx.vertx;
+import static net.consensys.cava.io.Base64.decodeBytes;
+import static net.consensys.cava.io.file.Files.copyResource;
+import static net.consensys.orion.acceptance.NodeUtils.joinPathsAsTomlListEntry;
+import static net.consensys.orion.http.server.HttpContentType.CBOR;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertEquals;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.cava.junit.TempDirectory;
+import net.consensys.orion.acceptance.NodeUtils;
+import net.consensys.orion.cmd.Orion;
+import net.consensys.orion.config.Config;
+import net.consensys.orion.http.server.HttpContentType;
+import net.consensys.orion.network.ConcurrentNetworkNodes;
+import net.consensys.orion.utils.Serializer;
+
+import java.nio.file.Path;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpClient;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+class PrivacyGroupAcceptanceTest {
+
+  static final String PK_1_B_64 = "A1aVtMxLCUHmBVHXoZzzBgPbW/wj5axDpW9X8l91SGo=";
+  static final String PK_2_B_64 = "Ko2bVqD+nNlNYL5EE7y3IdOnviftjiizpjRt+HTuFBs=";
+  static final String PK_3_B_64 = "6pFBKvouUVdoXRownMEHrRl8SbLJXIZC0EQOmC+WBlg=";
+
+  private Config firstNodeConfig;
+  private Config secondNodeConfig;
+  private Config thirdNodeConfig;
+  private ConcurrentNetworkNodes networkNodes;
+
+  Orion firstOrionLauncher;
+  Orion secondOrionLauncher;
+  Orion thirdOrionLauncher;
+  private Vertx vertx;
+  HttpClient firstHttpClient;
+  HttpClient secondHttpClient;
+  HttpClient thirdHttpClient;
+
+  @BeforeEach
+  void setUpTriNodes(@TempDirectory Path tempDir) throws Exception {
+
+    Path key1pub = copyResource("key1.pub", tempDir.resolve("key1.pub"));
+    Path key1key = copyResource("key1.key", tempDir.resolve("key1.key"));
+    Path key2pub = copyResource("key2.pub", tempDir.resolve("key2.pub"));
+    Path key2key = copyResource("key2.key", tempDir.resolve("key2.key"));
+    Path key3pub = copyResource("key3.pub", tempDir.resolve("key3.pub"));
+    Path key3key = copyResource("key3.key", tempDir.resolve("key3.key"));
+
+    String jdbcUrl = "jdbc:h2:" + tempDir.resolve("node2").toString();
```

Comment:
is there a reason one node is using h2? Is that part of the test?

Reply:
This is an artefact of the setup used here: https://github.com/PegaSysEng/orion/blob/master/src/acceptance-test/java/net/consensys/orion/acceptance/send/receive/DualNodesSendReceiveTest.java - which was the basis for this acceptance test.